test/ocsp_test.go

```diff
@@ -0,0 +1,563 @@
+
+	// Should not be connection refused since server will continue running.
+	_, err = nats.Connect(fmt.Sprintf("tls://localhost:%d", opts.Port),
+		nats.RootCAs(caCert),
+		nats.ErrorHandler(noOpErrHandler),
+	)
+	if err != nil {
+		t.Errorf("Unexpected error: %s", err)
+	}
+}
+
+func TestOCSPClient(t *testing.T) {
+	const (
+		caCert     = "configs/certs/ocsp/ca-cert.pem"
+		caKey      = "configs/certs/ocsp/ca-key.pem"
+		serverCert = "configs/certs/ocsp/server-cert.pem"
+		serverKey  = "configs/certs/ocsp/server-key.pem"
+	)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ocspr := newOCSPResponder(t, caCert, caKey)
+	ocspURL := fmt.Sprintf("http://%s", ocspr.Addr)
+	defer ocspr.Shutdown(ctx)
+
+	for _, test := range []struct {
+		name      string
+		config    string
+		certs     nats.Option
+		err       error
+		rerr      error
+		configure func()
+	}{
+		{
+			"OCSP Stapling makes server fail to boot if status is unknown",
+			`
+				port: -1
+
+				# Enable OCSP stapling with policy to honor must staple if present.
+				ocsp: true
+
+				tls {
+					cert_file: "configs/certs/ocsp/server-cert.pem"
+					key_file: "configs/certs/ocsp/server-key.pem"
+					ca_file: "configs/certs/ocsp/ca-cert.pem"
+					timeout: 5
+				}
+			`,
+			nats.ClientCert("./configs/certs/ocsp/client-cert.pem", "./configs/certs/ocsp/client-key.pem"),
+			nil,
+			nil,
+			func() {},
+		},
+		{
+			"OCSP Stapling ignored by default if server without must staple status",
+			`
+				port: -1
+
+				# Enable OCSP stapling with policy to honor must staple if present.
+				ocsp: true
+
+				tls {
+					cert_file: "configs/certs/ocsp/server-cert.pem"
+					key_file: "configs/certs/ocsp/server-key.pem"
+					ca_file: "configs/certs/ocsp/ca-cert.pem"
+					timeout: 5
+				}
+			`,
+			nats.ClientCert("./configs/certs/ocsp/client-cert.pem", "./configs/certs/ocsp/client-key.pem"),
+			nil,
+			nil,
+			func() { setOCSPStatus(t, ocspURL, serverCert, ocsp.Good) },
+		},
+		{
+			"OCSP Stapling honored by default if server has must staple status",
+			`
+				port: -1
+
+				ocsp: true
+
+				tls {
+					cert_file: "configs/certs/ocsp/server-status-request-url-cert.pem"
+					key_file: "configs/certs/ocsp/server-status-request-url-key.pem"
+					ca_file: "configs/certs/ocsp/ca-cert.pem"
+					timeout: 5
+				}
+			`,
+			nats.ClientCert("./configs/certs/ocsp/client-cert.pem", "./configs/certs/ocsp/client-key.pem"),
+			nil,
+			nil,
+			func() {
+				setOCSPStatus(t, ocspURL, "configs/certs/ocsp/server-status-request-url-cert.pem", ocsp.Good)
+			},
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			test.configure()
+			content := test.config
+			conf := createConfFile(t, []byte(content))
+			defer removeFile(t, conf)
+			s, opts := RunServerWithConfig(conf)
+			defer s.Shutdown()
+
+			nc, err := nats.Connect(fmt.Sprintf("tls://localhost:%d", opts.Port),
+				test.certs,
+				nats.RootCAs(caCert),
+				nats.ErrorHandler(noOpErrHandler),
+			)
```

Comment:
Commit from the nats.go that shows how to verify OCSP responses in the client: https://github.com/nats-io/nats.go/commit/50e25c411c42c6a6998252bee9ce50be89137077#diff-1d2fb9af677137c0182213770500dd5625a51ba7286301cb11d0197b21e0f948R1755